mlflow/models/utils.py

```diff
@@ -264,3 +276,313 @@ def plot_lines(data_series, xlabel, ylabel, legend_loc=None, line_kwargs=None):
+                    " {0}".format(str(e))
+                )
+        if not isinstance(pfInput, pd.DataFrame):
+            raise MlflowException(
+                "Expected input to be DataFrame or list. Found: %s" % type(pfInput).__name__
+            )
+
+    if input_schema.has_input_names():
+        # make sure there are no missing columns
+        input_names = input_schema.input_names()
+        expected_cols = set(input_names)
+        actual_cols = set()
+        if len(expected_cols) == 1 and isinstance(pfInput, np.ndarray):
+            # for schemas with a single column, match input with column
+            pfInput = {input_names[0]: pfInput}
+            actual_cols = expected_cols
+        elif isinstance(pfInput, pd.DataFrame):
+            actual_cols = set(pfInput.columns)
+        elif isinstance(pfInput, dict):
+            actual_cols = set(pfInput.keys())
+        missing_cols = expected_cols - actual_cols
+        extra_cols = actual_cols - expected_cols
+        # Preserve order from the original columns, since missing/extra columns are likely to
+        # be in same order.
+        missing_cols = [c for c in input_names if c in missing_cols]
+        extra_cols = [c for c in actual_cols if c in extra_cols]
+        if missing_cols:
+            message = "Model is missing inputs {0}.".format(missing_cols)
+            if extra_cols:
+                message += " Note that there were extra inputs: {0}".format(extra_cols)
+            raise MlflowException(message)
+    elif not input_schema.is_tensor_spec():
+        # The model signature does not specify column names => we can only verify column count.
+        num_actual_columns = len(pfInput.columns)
+        if num_actual_columns < len(input_schema.inputs):
+            raise MlflowException(
+                "Model inference is missing inputs. The model signature declares "
+                "{0} inputs  but the provided value only has "
+                "{1} inputs. Note: the inputs were not named in the signature so we can "
+                "only verify their count.".format(len(input_schema.inputs), num_actual_columns)
+            )
+
+    return (
+        _enforce_tensor_schema(pfInput, input_schema)
+        if input_schema.is_tensor_spec()
+        else _enforce_col_schema(pfInput, input_schema)
+    )
+
+
+def validate_schema(data: DataInputType, expected_schema: Schema) -> DataInputType:
+    """
+    Validate that the input data schema matches expected schema.
+
+    :param data: Input data to be validated. Supported types are:
+
+                 - A Pandas DataFrame.
+
+                 - Pandas Series
+
+                 - Pandas DataFrame
+
+                 - Pandas Series
+
+                 - Numpy ndarray
+
+                 - scipy.sparse.csc_matrix
+
+                 - scipy.sparse.csr_matrix
+
+                 - List[Any]
+
+                 - Dict[str, Any]
+
+                 - list
```

Comment:
```suggestion
                 - pandas.DataFrame
                 - pandas.Series
                 - numpy.ndarray
                 - scipy.sparse.csc_matrix
                 - scipy.sparse.csr_matrix
                 - List[Any]
                 - Dict[str, Any]
```

Reply:
I changed this so that the auto-generated doc renders it as bullet points. If we don't add blank lines in between it just appends after each other.

Reply:
I built doc with the suggested code and it looks like this:

<img width="1094" alt="image" src="https://user-images.githubusercontent.com/17039389/186333290-923852f1-1db5-4c3c-9b0f-db80648d9377.png">

Reply:
We need a blank space before the first item but don't between each item.